homeassistant/components/media_extractor/__init__.py

```diff
@@ -128,9 +131,12 @@ def stream_selector(query):
                 raise MEQueryException() from err
 
             if "formats" in requested_stream:
-                best_stream = requested_stream["formats"][
-                    len(requested_stream["formats"]) - 1
-                ]
+                widthmax = 0
```

Comment:
I was the one who updated this last time, and I am no expert here so maybe together we can figure this out!

I tried to read the ytdlp docs, but they are made for command line, so I was looking a lot at the code of ytdlp, and my conclusion was that the best format was the one that was last. Maybe I didn't look good enough or I had to look at another place, but please correct me if I am wrong :)

Reply:
My code ensures to select the stream with video and audio (there are some streams with only audio or only video) and the one with highest width, excluding the ones that are not playable on media_player entity (like manifest.googlevideo.com)

Reply:
I'd rather do a bit more research on how ytdlp works